### PR TITLE
[Remove Stream/Optional usage] LPS-178247 remove_stream_and_optional_in_commerce_qualifier_service

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina, Dante,

This is for ticket https://issues.liferay.com/browse/LPS-178247
Thanks for checking!

### CODE REVIEW

**Comment by zhangyan0701** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-qualifier-service/src/main/java/com/liferay/commerce/qualifier/service/impl/CommerceQualifierEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -406,13 +406,16 @@ private GroupByStep _getGroupByStep(
...
-		String[] allowedTargetKeys = Stream.of(
-			sourceCommerceQualifierMetadata.getAllowedTargetKeysArray()
-		).flatMap(
-			Stream::of
-		).toArray(
-			String[]::new
-		);
+		List<String> allowedTargetKeysList = new ArrayList<>();
+
+		for (String[] row :
```
Comment (dantewang) [inline modules/apps/commerce/commerce-qualifier-service/src/main/java/com/liferay/commerce/qualifier/service/impl/CommerceQualifierEntryLocalServiceImpl.java:411]: @jeffwu0724 Why is this called "row"?

**Comment by dantewang** [comment]
ci:stop

**Comment by dantewang** [comment]
@jeffwu0724 Please see comment and resend.

**Pull request closed without merging**